### PR TITLE
Cancel Booking API implementation

### PR DESCRIPTION
**Pull request by Ravichandran-TK**

### CODE REVIEW

File: mailer/booking_mailer.go [hunk 1]
```diff
@@ -11,22 +12,27 @@ func BookingMailer(bookingId int16, reminder bool) {
 	particitpants := model.GetBookingParticipantsDetailsByBookingId(bookingId)
 
 	recipients := make([]*model.Recipient, 0)
-
+	bookingData, _ := model.FetchBooking(bookingId)
```
Comment (maheshnanubala) [inline mailer/booking_mailer.go:15]: donot ignore errors

File: mailer/booking_mailer.go [hunk 2]
```diff
@@ -11,22 +12,27 @@ func BookingMailer(bookingId int16, reminder bool) {
...
 	for _, participant := range particitpants {
 		recipient := new(model.Recipient)
 		recipient.Name = participant.UserName
 		recipient.Email = participant.UserEmail
 		recipients = append(recipients, recipient)
 	}
+	commonEmails := strings.SplitAfter(bookingData.CommonEmails, ",")
+	for _, commonEmail := range commonEmails {
```
Comment (maheshnanubala) [inline mailer/booking_mailer.go:23]: return the error

File: mailer/booking_mailer.go [hunk 3]
```diff
@@ -11,22 +12,27 @@ func BookingMailer(bookingId int16, reminder bool) {
...
+		commonRecipient := new(model.Recipient)
+		commonRecipient.Name = commonEmail
+		commonRecipient.Email = commonEmail
+		recipients = append(recipients, commonRecipient)
+	}
 
 	const (
 		layoutISO  = "2006-01-02"
 		layoutUS   = "Monday, Jan 2 2006"
 		timeLayout = "15:04 PM"
```
Comment (maheshnanubala) [inline mailer/booking_mailer.go:33]: move to constants folder

File: mailer/cancel_booking_mailer.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
+package mailer
+
+import (
+	"os"
+	"strings"
+	"workspace_booking/model"
+)
+
+func CancelMailer(bookingId int16) {
+	templatePath := "/text/cancelbooking-template.html"
```
Comment (maheshnanubala) [inline mailer/cancel_booking_mailer.go:10]: constant

File: mailer/cancel_booking_mailer.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		recipient.Name = commonEmail
+		recipient.Email = commonEmail
+		recipients = append(recipients, recipient)
+	}
+
+	const (
+		layoutISO  = "2006-01-02"
+		layoutUS   = "Monday, Jan 2 2006"
+		timeLayout = "15:04 PM"
+	)
```
Comment (maheshnanubala) [inline mailer/cancel_booking_mailer.go:34]: constant

File: mailer/cancel_booking_mailer.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		"EndTime":           EndTime,
+		"Date":              formatDate,
+		"City":              bookingData.CityName,
+		"Building":          bookingData.BuildingName,
+		"Floor":             bookingData.FloorName,
+		"WorkspaceName":     bookingData.BookingWorkspace[len(bookingData.BookingWorkspace)-1].WorkspaceName,
+		"WorkspaceCapacity": bookingData.BookingWorkspace[len(bookingData.BookingWorkspace)-1].WorkspaceCapacity,
+		"BaseUrl":           baseUrl,
+		"UserName":          bookingData.UserName,
+	}
```
Comment (maheshnanubala) [inline mailer/cancel_booking_mailer.go:65]: make it as a common function

**Review by maheshnanubala — CHANGES_REQUESTED** [review]

File: mailer/booking_mailer.go (hunk 1 above)
Comment (Ravichandran-TK) [reply mailer/booking_mailer.go]: done the changes.

File: mailer/booking_mailer.go (hunk 2 above)
Comment (Ravichandran-TK) [reply mailer/booking_mailer.go]: its index value not a error

File: mailer/booking_mailer.go (hunk 3 above)
Comment (Ravichandran-TK) [reply mailer/booking_mailer.go]: done the changes

File: mailer/cancel_booking_mailer.go (hunk 4 above)
Comment (Ravichandran-TK) [reply mailer/cancel_booking_mailer.go]: done the changes

File: mailer/cancel_booking_mailer.go (hunk 5 above)
Comment (Ravichandran-TK) [reply mailer/cancel_booking_mailer.go]: done the changes

File: mailer/cancel_booking_mailer.go (hunk 6 above)
Comment (Ravichandran-TK) [reply mailer/cancel_booking_mailer.go:65]: as of now we cant move this to common function

File: mailer/booking_mailer.go [hunk 7]
```diff
@@ -2,30 +2,38 @@ package mailer
...
 	"os"
+	"strings"
+	"workspace_booking/config"
 	"workspace_booking/model"
 )
 
 func BookingMailer(bookingId int16, reminder bool) {
 
-	templatePath := "/text/email-template.html"
+	templatePath := config.GeBookingTemplatePath()
```
Comment (maheshnanubala) [inline mailer/booking_mailer.go:12]: not config, constant variabe is enough

File: mailer/cancel_booking_mailer.go (hunk 6 above)
Comment (maheshnanubala) [reply mailer/cancel_booking_mailer.go:65]: U need to split this in to multiple functions

**Review by maheshnanubala — APPROVED** [review]
Please make these change in the coming PR

**Pull request merged**